src/libs/ReportUtils.ts

```diff
@@ -1657,6 +1657,36 @@ function getRoomWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
     return welcomeMessage;
 }
 
+function getWelcomeMessage(report: OnyxEntry<Report>): WelcomeMessage {
+    const welcomeMessage: WelcomeMessage = {showReportName: true};
+    if (isChatRoom(report)) {
+        return getRoomWelcomeMessage(report);
+    }
+
+    if (isPolicyExpenseChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartOne');
+        welcomeMessage.phrase2 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartTwo');
+        welcomeMessage.phrase3 = Localize.translateLocal('reportActionsView.beginningOfChatHistoryPolicyExpenseChatPartThree');
+        return welcomeMessage;
+    }
+
+    if (isSelfDM(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySelfDM');
+        return welcomeMessage;
+    }
+
+    if (isSystemChat(report)) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistorySystemDM');
+        return welcomeMessage;
+    }
+
+    if (!(isChatRoom(report) || isPolicyExpenseChat(report) || isSelfDM(report) || isInvoiceRoom(report) || isSystemChat(report))) {
+        welcomeMessage.phrase1 = Localize.translateLocal('reportActionsView.beginningOfChatHistory');
+    }
+    
+    return welcomeMessage;
```

Comment:
The return should be inside the if condition. If no condition is met we should return `undefined`